api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -189,11 +201,26 @@ private Order saveOrder(Order order, OrderContext orderContext, boolean isRetros
 			}
 			order.setCareSetting(careSetting);
 		}
-		
+	}
+
+	private void failOnOrderTypeMismatch(Order order) {
 		if (!order.getOrderType().getJavaClass().isAssignableFrom(order.getClass())) {
 			throw new OrderEntryException("Order.type.class.does.not.match", new Object[] {
-			        order.getOrderType().getJavaClass(), order.getClass().getName() });
+					order.getOrderType().getJavaClass(), order.getClass().getName() });
+		}
+	}
+
+	private Order saveOrder(Order order, OrderContext orderContext, boolean isRetrospective) {
```

Comment:
@Nuwantha currently I cannot access jira and adapt the issue. But it should be clear from the method names what the method does and since the code snippet in the issue of how it should look in the end is correct and should also help you. furthermore please put the private methods after the method `saveOrder` as stated in the issue. this makes reading of the code easier since it follow the natural read from top to bottom and simply flows :)